### PR TITLE
Migration of task-submission-viewer component to Angular

### PR DESCRIPTION
**Pull request by SamAnast**
# Description
Migrated task-submisison-viewer component from AngularJS to Angular.

## Type of change

Upgrade to new framework

# Checklist:

- [x] My changes generate no new warnings

If you have any questions, please contact @macite or @jakerenzella.

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -158,5 +158,8 @@
     "@schematics/angular:component": {
       "style": "scss"
     }
+  },
+  "cli": {
+    "analytics": "72a6ca29-562e-4020-abd5-0c7dd3541311"
```
Comment (jakerenzella) [inline angular.json:163]: What is this? Was it automatically added? 🤔

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          <i class="fa fa-clock-o fa-2x"></i>
+          <p>
+            Submission is still being processed
+            <p class="small text-muted">
+              Please check again shortly
+            <!-- </p> -->
+          </p>
+      </div><!--/pdf-in-process-->
+      <div [hidden]="!loadingDetails(task)">
+          <i class="fa fa-spinner fa-pulse fa-2x"></i>
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.html:17]: Please use the Material Spinner here, not the old bootstrap one.

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  ngOnInit(): void {
+  }
+
+  notSubmitted(task): boolean {
+    return (!task.has_pdf && (!task.processing_pdf))
+  }
+
+  loadingDetails(task): boolean {
+    return (task.needsSubmissionDetails())
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts:30]: Do we need this method? Can we just use the method on task in the directive?

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    changes.task.previousValue;
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts:34]: What are we doing with previousValue?

**Review by jakerenzella — CHANGES_REQUESTED** [review]
Overall looks good, just a few small things. Nice work 🎆

File: angular.json (hunk 1 above)
Comment (SamAnast) [reply angular.json:163]: Yeah It must have been. Should I remove it?

Comment (jakerenzella) [reply angular.json:163]: Yes, I'll need to review if it's needed and what it's doing.

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts (hunk 3 above)
Comment (SamAnast) [reply src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts]: The above has been removed

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts (hunk 4 above)
Comment (SamAnast) [reply src/app/tasks/task-submission-viewer/task-submission-viewer.component.ts]: Removed

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.html [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
     </div><!--/no-pdf-->
-    <div ng-show="task.processing_pdf && ! loadingDetails(task)">
+    <div [hidden]="!(task.processing_pdf && ! task.needsSubmissionDetails())">
         <i class="fa fa-clock-o fa-2x"></i>
         <p>
           Submission is still being processed
-          <p class="small text-muted">
-            Please check again shortly
-          </p>
+          <p class="small text-muted">Please check again shortly
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.html:11]: Just noticed this is a bootstrap style, would want something similar in https://material.angular.io/guide/typography

**Review by jakerenzella — CHANGES_REQUESTED** [review]
Looks pretty close, just that change and I'll test this locally myself

**Comment by jakerenzella** [comment]
@SamAnast This will need to be rebased with the new inbox-redesign updates

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.html [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
     </div><!--/no-pdf-->
-    <div ng-show="task.processing_pdf && ! loadingDetails(task)">
+    <div [hidden]="!(task.processing_pdf && ! task.needsSubmissionDetails())">
         <i class="fa fa-clock-o fa-2x"></i>
         <p>
           Submission is still being processed
-          <p class="small text-muted">
-            Please check again shortly
-          </p>
+          <p class = "muted-text mat-small">Please check again shortly
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.html:11]: These are still bootstrap styles that need to be migrated to Angular Material Typography

File: src/app/tasks/task-submission-viewer/task-submission-viewer.component.html [hunk 7]
```diff
@@ -1,29 +1,27 @@
 <div class="panel-body task-submission-viewer">
-  <div class="large-notice-block" ng-show="notSubmitted(task) || task.processing_pdf || loadingDetails(task)">
-    <div ng-show="notSubmitted(task) && ! loadingDetails(task)">
+  <div class="large-notice-block" [hidden]="!(notSubmitted(task) || task.processing_pdf || task.needsSubmissionDetails())">
+    <div [hidden]="!(notSubmitted(task) && ! task.needsSubmissionDetails())">
       <i class="fa fa-upload fa-2x"></i>
```
Comment (jakerenzella) [inline src/app/tasks/task-submission-viewer/task-submission-viewer.component.html:4]: Same here, should be mat-icon

**Review by jakerenzella — CHANGES_REQUESTED** [review]

**Comment by ThomasLaing** [comment]
@jakerenzella Is there anything outstanding, or can this pull request be merged?

**Comment by jakerenzella** [comment]
I've taken a look at this now, I'm not actually sure this component is used anywhere in the application? @macite?

**Comment by macite** [comment]
Not sure @jakerenzella. I just noticed you had reviewed this before so suggested @ThomasLaing get in touch.

@ThomasLaing have you identified where this component is used?

**Comment by macite** [comment]
No progress...

**Pull request closed without merging**